assets/elm/Data/Board.elm

```diff
@@ -128,8 +155,13 @@ nextSnakeAndApples time snake apples =
 
 
 score : Board -> Int
-score { snake } =
-    List.length snake.body
+score { currentPlayerId, snakes } =
+    case Dict.get currentPlayerId snakes of
+        Nothing ->
+            0
```

Comment:
Haha :) fair enough :) the only trouble with that is that the `maybe` function is actually "more efficient" ;-) (ie if it's `Nothing` it skips the mapping entirely. The "joe's mental model with pipes" version has to map in every case (even if the map **is** simply the `identity` function on the `Nothing` value, it's still more computation being done).

I think all these things start to matter when there's lots of code, but they don't really matter in the small much.

All things told, I think I prefer this, TBH:

```
score : Board -> Int
score { currentPlayerId, snakes } =
    let
        maybeCurrentPlayerSnake = Dict.get currentPlayerId snakes
        bodyLength = List.length << .body
    in
        maybe 0 bodyLength maybeCurrentPlayerSnake
```